### ISSUE TITLE
Make App Headings Consistent - Audit

### ISSUE THREAD

**Issue by kellylein** [department-of-veterans-affairs/va-mobile-app#4603]
# Project Status Overview

### Project Objective:

- Right now, our heading use is not consistent throughout the app, and many of the new features we're working on have few (if any) headings.
- This epic tracks auditing the application for where there should be headings. The output of this epic will be another epic which specifies exactly where headings should be added.
 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
| Work not prioritized for Mobile to work on  | Mobile POs   | TBD |
|  |   |
|  |   |

</details>

### Project Timeline
<details>
<summary>Milestones</summary>

|Step | Timing | 
|-------| ---------| 
|Sprint 0 - Discovery |  Nov 2022   |
|Step 1 - Strategy |   |
|Step 2 - Design  |   |
|Step 3 - Building / Implementing |   |
| 100% Release |  |
|Step 4 - Post Verification|   |
</details>

### Monthly Update 
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|Sep '23| Not Started | Epic updated to new template; project included in Q4 planning | Work not prioritized |
|Oct '23| Not Started | No update  | Work not prioritized |
|Nov'23| Not Started | Work was originally planned for Sprint 84 however it was pulled out of Sprint 84 due to capacity and resource issues. It will be revisited in future sprints for consideration.  | Work not prioritized |
</details>

# Detailed Project Scope 

### Problem Statement

- Right now, our heading use is not consistent throughout the app, and many of the new features we're working on have few (if any) headings.

- Current Experience:  Pretend that the only thing you could see when you landed on a page were the headings. Could you figure out 1) what information was on the page (is the thing I'm looking for on this page?) and 2) the fastest way to get to that information (to find X, I will look at the stuff under Y heading) - [an example slideshow](https://docs.google.com/presentation/d/1Ei1-VsVyPAuuF0Jf3YAzDk7FqUFnULxh5vuO3-XaHr8/edit?usp=sharing)

### Assumptions 
<details>
<summary> List of Assumptions </summary>

- Headings matter for screen reader users. There’s a lot of literature on this, including several WCAG success criterion and various web postings:
      - From [WCAG](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html): Descriptive headings are especially helpful for users who have disabilities that make reading slow and for people with limited short-term memory. These people benefit when section titles make it possible to predict what each section contains.
      - From [an A11y Project post](https://www.a11yproject.com/posts/how-to-accessible-heading-structure/) (about websites, the principles apply to mobile): Headings are the backbone of the content of a page. A visitor should be able to scan the webpage using headings to get a good impression of its content. Heading levels have meaning, especially for screen reader users .... headings [represent] the page content similar to the index of a book: easily scannable and semantic. 
- Headings matter to the mobile team. A few relevant snippets from our current [accessibility standards](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/testing/Accessibility%20Test%20Plan.md): 
      - We will “programmatically specify ARIA level for all text headings in React Native.”
      - “Screen reader should provide a breadcrumb like page read”
       -  “Headings and labels design should describe topic or purpose and be programmatically determined as such.”
</details>

### Risks 
<details>
<summary> List of Risk </summary>

- We spend time & energy bringing current screens to a good standard, but fail to future-proof and end up in a similar place for new features in a year
- The modifications we make are up to our standards but don’t improve the experience for Veterans (screen reader users, etc).

</details>

### Definition of Done 
<details>
<summary>Following needs to be true</summary>

|Item | Completed | 
|-------| ---------|
| Full Audit of pages in the app and what headings are being used | have headings more ‘built-in’ to our design library and engineering components, so future us gets a really good basic approach to headings for free
 | 
| Come up with clear guidelines for heading structure: What should have a heading (which sections) and how headings should be written (what content/style for headings)| 
 | Add automated checks for headings | |
 | Document in all the places (design library, documentation site)| |

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| | | 
| | | 

</details>

### Important Project Links 
<details>
<summary>Links</summary>

- [Github](url)
- [Product Brief](url)
- [Product Canvas](url)
- Design 
   - [Discovery](url)
   - [Mockups](url)
   - [Final Design](url)
   - [Figma File](url)
- Research
   - [Document 1](url) 
   - [Document 2 ](url) 
-Analytics
   - [Document 1](url) 
</details>

### Team Members

<details>
<summary>Contact List</summary>

- **OCTO Product Leads:** Rachel Han and Ryan Thurlwell
- **Ad Hoc PM:** 
- **Engineering:**
- **Design:** Brea Blackwelder
- **External Team 1 & Role:** Martha Wilkes
- **External Team 1 & Role:**

</details>

### OKR and Analytic Metrics 
<details>
<summary>Metrics</summary>

- Internal: 
     - QA team attests that app headings meet our accessibility standards
     - % of screens with compliant headings (vs non-compliant headings) increases
- External: 
    - Qualitative:
        - Positive qualitative feedback in usability sessions and app store reviews
    - Quantitative:
        - % of eligible users can successfully complete a task with first attempt goes up on tasks where we implement the headings
        - Quicker completion of task on tasks where we implement the headings (all users, screen reader users)

|What we are measuring | Why | Event Name 
|-------| ---------| ----- |
| | | |
| | | |

</details>

### Questions 
<details>
<summary>Open Questions</summary>

</details>

<details>
<summary>Closed Questions</summary>

| Question |  Answer |
|--------|----------|
| | |
| | |

</details>

### Change / Decision Log 

<details>
<summary>Change & Decision Log Details </summary>

| Change | Rationale | Decision Maker |  Date Change |
|-------|------- | -------- | -------- | 
| | | | |
| | | | |

</details>

# Checklist

<details>
  <summary>Toggle Product Checklist</summary>

**Project Kickoff & Requirements Gathering**
   - [ ] Project Kickoff session
   - [ ] Initiative Scope finalized
   - [ ] Link Product Brief - [Sprint 0 ](https://docs.google.com/document/d/1WHDJ1iCt9J-acLx8_DSL4tkRKoVYqJGaRUqXuFf7ceg/edit# )
   - [ ] Link to Implementation Strategy
       - Implementation Plan
       - UAT Plan
       - Release Plan
       - Metrics Identified (what metrics are we capturing, how do they tie into stakeholder OKRs, etc)

**Pre-Production Readiness**
- [ ] Associate Readiness
  - [ ] Update product guide
  - [ ] Coordinate with contact center support teams
- [ ] What’s New Content
- [ ] App Store/In App messaging (if applicable)
- [ ] Coordinate with Release coordinator
- [ ] App store Content changes required? Y/N
- [ ] App store images needed? Y/N
- [ ] QA Signed off 
- [ ] UAT complete
- [ ] Obtain Go/No Go decision

**Feature Go Live**
- [ ] Ensure Waygate/Feature toggle release ticket submitted w/engineering & QA 
  - **Template: Turn on Waygate/Turn off feature flag**
- [ ] Coordinate with content & release coordinator if what’s new in app is required
</details>

<details>
  <summary>Toggle Team Collaboration Checklist</summary>

- [ ] Design Intent & Feasibility discussions
- [X] Research needed? No
- [ ] UAT Needed? Yes/No
- [ ] Ticket creation 
- [ ] Demo feature (mid check, end of development)
</details>

<details>
  <summary>Toggle Design/UX Checklist</summary>

- [ ] Review problem/opportunity statement
- [ ] Add questions and assumptions to product brief
- [ ] Add research links to product brief
- [ ] Document high-level research/design plans
- [ ] Create lo/hi fi wireframes & user flows if applicable
- [ ] Add links to Epic & applicable GitHub folders
- [ ] Socialize designs
- [ ] Sign off received
- [ ] Complete Design QA
- [ ] Does it require component review?
- [ ] Accessibility review needed?

</details>

<details>
  <summary>Toggle Content Checklist</summary>

- [ ] Review product brief and surface any questions, assumptions & risks
- [ ] Participate in project discovery and kickoff activities/ceremonies
- [ ] Complete comparative analysis/content research activities for net-new content
- [ ] Review current content and determine where app may need to differ and potential improvements to share with web
- [ ] Review past VA research and decisions documentation
- [ ] Support Research & UX Design in research sessions and synthesis
- [ ] Make content recommendations for lo-/hi-fi wireframes in collaboration with UX Design
- [ ] Collaborate with Sitewide Content team for alignment, improvements, and sign-off
- [ ] Participate in FE hand off
- [ ] Complete content QA
- [ ] Write copy for What's New In App
- [ ] Write copy for app stores' What's New sections.
- [ ] Write copy for updated screenshots in app stores 

</details>

<details>
  <summary>Toggle Engineering Checklist</summary>

- [ ] Are UI designs/ specs available/ready? If yes, they should be attached to  attach specs. (We are assuming the design specs have been approved)
- [ ] Are there any deadlines that need to be met?
- [ ] What is the roll out plan? (Do we need a feature toggle? is this going to be available to every user immediately or do we want to roll out to a few users first)
- [ ] Does this project require api integration? if yes, what api endpoint are we integrating with and what is the expected response body?
- [ ] Do we want to capture analytics for this projects (i.e. update GA, add logs to grafana/datadog etc)?If yes, what metrics would we like to capture?

</details>

<details>
  <summary>Toggle QA Checklist</summary>

- [ ] Create test plan
  -  **Template: QA Test Plan**
- [ ] Identify test data needs
- [ ] Gotten & incorporated test plan review
- [ ] QA Complete
- [ ] Waygate tasks complete
- [ ] Added new feature cases to the RC in TestRail
- [ ] Added feature cases to appropriate section of active cases in TestRail
- [ ] Spun up a (future) UI automation ticket for new feature
</details>

**Comment by bischoffa** [comment]
Planned for Sprint 84 but pushed out because of PM capacity and other priorities.

**Comment by brea11y** [comment]
The app headings audit has been completed and the full list of results can be found [in this Google sheet](https://docs.google.com/spreadsheets/d/1KnR9Isdw8e3aBp5WhOFk7vaR4p6v4-wpYIO4GsEk4Go/edit?usp=sharing). I have also created [ticket #7677](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7677) for Misty with a list of app headings that need content's review.
